### PR TITLE
test: unit tests for jax logging components

### LINKED ISSUE #525: [TEST] Jax Loggers
### What do you want to test?
Jax logger components

### Outline of test structure
* Unit tests
* Test components and hooks

### Definition of done
Passing checks, cover all hooks, edge cases considered

### Mandatory checklist before making a PR
* [ ] The success criteria laid down in “Definition of done” are met.
* [ ] Test code is documented - docstrings for methods and classes, static types for arguments.
* [ ] Documentation is updated - README, CONTRIBUTING, or other documentation.

### PR DESCRIPTION
**Pull request by mmorris44**
## What?
Unit tests for jax logging components
## Why?
Component currently not tested. Needs to be robust for jax Mava release
## How?
- Test `Logger` component
- Fixed typo in `mocks.py`
## Extras
Closes #525

### CODE REVIEW

File: tests/jax/components/building/loggers_test.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    """Pytest fixture for TestLogger.
+
+    Args:
+        test_logger_factory: factory to use in logger config.
+
+    Returns:
+        Default TestLogger.
+    """
+    test_logger = TestLogger(test_logger_factory)
+    return test_logger
```
Comment (sash-a) [inline tests/jax/components/building/loggers_test.py:78]: Just `return TestLogger(test_logger_factory)` here

File: tests/jax/mocks.py [hunk 2]
```diff
@@ -473,7 +473,7 @@ class MockLoggerConfig:
     logger_param_1: int = 2
 
 
-class MockLogerClass:
```
Comment (sash-a) [inline tests/jax/mocks.py:476]: Logers > Loggers :joy:

**Review by sash-a — CHANGES_REQUESTED** [review]
Very minor change, but otherwise looks good :fire:

**Review by sash-a — DISMISSED** [review]
Thanks Matthew :+1:

**Review by KaleabTessera — COMMENTED** [review]
Thanks @mmorris44 . 

It looks good, but I was wondering if we could pass in the logger as a component to the builder and test it like that? Without manually calling `on_building_executor_logger`. That way we are testing that the builder correctly handles the logger. 

I did something similar with the data server tests - https://github.com/instadeepai/Mava/blob/153b3f6cab8150a9acec5f65ae9366f22207598f/tests/jax/mocks.py#L817.

**Comment by mmorris44** [comment]
> It looks good, but I was wondering if we could pass in the logger as a component to the builder and test it like that? Without manually calling `on_building_executor_logger`. That way we are testing that the builder correctly handles the logger.

I did think about doing this, but decided to rather manually call the hooks. This aligns with our goal of testing components in isolation. What you're describing above feels to me like a builder integration test, not a `Logger` test. Perhaps we should test both?

File: tests/jax/components/building/loggers_test.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    ):
+        """Create TestLogger
+
+        Args:
+            test_logger_factory: factory to use in the logger config.
+        """
+        logger_config = LoggerConfig()
+        logger_config.logger_factory = test_logger_factory
+        logger_config.logger_config = {
+            "trainer": {"time_stamp": "trainer_config"},
```
Comment (RuanJohn) [inline tests/jax/components/building/loggers_test.py:27]: Very minor, but can we make these `trainer_logger_config`, `executor_logger_config` and `evaluator_logger_config` instead? It makes reading this a littel bit easier.

Comment (mmorris44) [reply tests/jax/components/building/loggers_test.py]: Agreed, done :)

**Review by RuanJohn — APPROVED** [review]
Thank you @mmorris44 :muscle:

**Pull request merged**